src/metabase/search/appdb/index.clj

```diff
@@ -45,6 +52,46 @@
    (when (and table-name (exists? table-name))
      (t2/query (sql.helpers/drop-table table-name)))))
 
+(defn- existing-indexes []
+  (map (comp keyword :table_name)
+       (t2/query {:select [:table_name]
+                  :from   :information_schema.tables
+                  :where  [:like :table_name "search_index__%"]})))
+
+(defn- sync-tables [old-metadata new-metadata]
+  ;; Oh dear, we get the raw setting. Save a little bit of overhead by no keywordizing the keys.
+  (let [old-metadata                         (json/parse-string old-metadata)
+        {:strs [active-table pending-table]} (json/parse-string new-metadata)
+        ;; implicitly clear the pending table if we just activated it
+        pending-table                        (when (not= active-table pending-table) pending-table)]
+    (reset! *active-table* (some-> active-table keyword))
+    (reset! *pending-table* (some-> pending-table keyword))
+    ;; Clean up any tables not referenced by the previous or current configuration
```

Comment:
The hope here is that all other server sync the settings between any two mutations.

This is reasonable when we're re-indexing in the steady state, but rolling deployments or use of the re-indexing APIs could violate this assumption and cause rugs to get pulled out from underneath some of the servers.